modules/programs/kitty.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.kitty;
+
+  eitherStrBoolInt = with types; either str (either bool int);
+
+  optionalPackage = opt:
+    optional (opt != null && opt.package != null) opt.package;
+
+  fontType = (import ../lib/font-type.nix { inherit lib; }).fontType;
+
+  toKittyConfig = generators.toKeyValue {
+    mkKeyValue = key: value:
+    let
+      value' =
+        if isBool value then (if value then "yes" else "no")
+        else toString value;
+    in
+      "${key} ${value'}";
+  };
+
+  toKittyKeybindings = generators.toKeyValue {
+    mkKeyValue = key: command: "map ${key} ${command}";
+  };
+in
+
+{
+  options.programs.kitty = {
+    enable = mkEnableOption "Kitty terminal emulator";
+
+    config = mkOption {
```

Comment:
This option should be called `settings` instead for consistency with other modules and avoiding the already very overloaded `config` name.